website/docs/r/xray_encryption_config.html.markdown

```diff
@@ -0,0 +1,72 @@
+---
+subcategory: "XRay"
+layout: "aws"
+page_title: "AWS: aws_xray_encryption_config"
+description: |-
+    Creates and manages an AWS XRay Encryption Config.
+---
+
+# Resource: aws_xray_encryption_config
+
+Creates and manages an AWS XRay Encryption Config.
+
+~> **NOTE:** Removing this resource from Terraform has no effect to the encryption configuration within X-Ray.
+
+## Example Usage
+
+```hcl
+resource "aws_xray_encryption_config" "example" {
+  type = "NONE"
+}
+```
+
+## Example Usage with KMS Key
+
+```hcl
+resource "aws_kms_key" "example" {
+  description             = "Some Key"
+  deletion_window_in_days = 7
+
+  policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Id": "kms-tf-1",
+  "Statement": [
+    {
+      "Sid": "Enable IAM User Permissions",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "*"
+      },
+      "Action": "kms:*",
+      "Resource": "*"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_xray_encryption_config" "example" {
+  type   = "KMS"
+  key_id = aws_kms_key.example.arn
+}
+```
+
+## Argument Reference
+
+* `type` - (Required) The type of encryption. Set to `KMS` to use your own key for encryption. Set to `NONE` for default encryption.
+* `key_id` - (Optional) An AWS KMS customer master key (CMK). can be the Key ID, Key ARN or Alias. 
```

Comment:
Does the API always return the ARN of the target key? If so, we should not document the other values that will "unexpectedly" get canonicalized and show a difference.

Reply:
will check, i always miss this

Reply:
yep, it returns ARN. i'll add validation and amend docs.